src/index.ts

```diff
@@ -12,45 +10,169 @@ export async function init(code: BufferSource) {
+
+    let errMessage : string = "";
+    if (slice[0] !== OK_MARKER) {
+        errMessage = loadWasmBufferError(result);
+    }
+
+    wasmBufferFree(buf);
+    wasmBufferFree(result);
+
+    if (errMessage.length > 0) {
+        throw errMessage;
+    }
+
+    return slice.slice(1);
+}
+
+// Decodes the provided svm encoded spwan data
+export function decodeSpawnData(data: Uint8Array) : JSON {
+    const buf = newWasmBuffer({data: binToString(data)});
+    const result = call("wasm_decode_spawn", buf);
+    const json = loadWasmBufferDataAsJson(result);
+    wasmBufferFree(buf);
+    wasmBufferFree(result);
+    return json;
+}
+
+// Encodes the provided call account data
+export function encodeCallData(data:CallData) : Uint8Array {
+    const buf = newWasmBuffer(data);
+    const result = call("wasm_encode_call", buf);
+    const len = wasmBufferLength(result);
+    const slice = wasmBufferDataSlice(result, 0, len);
+
+    let errMessage : string = "";
+    if (slice[0] !== OK_MARKER) {
+        errMessage = loadWasmBufferError(result);
+    }
+
+    wasmBufferFree(buf);
+    wasmBufferFree(result);
+
+    if (errMessage.length > 0) {
+        throw errMessage;
+    }
+
+    return slice.slice(1);
+}
+
+// Decodes the svm encoded call account data
+export function decodeCallData(bytes: Uint8Array): JSON {
+    const buf = newWasmBuffer({ data: binToString(bytes) });
+    const result = call("wasm_decode_call", buf);
+    const json = loadWasmBufferDataAsJson(result);
+    wasmBufferFree(buf);
+    wasmBufferFree(result);
+    return json;
+}
+
+// Encodes data provided in object and returns the encoded data in a json object
+export function encodeInput(object: any) : JSON {
+    const buf = newWasmBuffer(object);
+    const result = call("wasm_encode_inputdata", buf);
+    const encoded = loadWasmBufferDataAsJson(result);
+    wasmBufferFree(buf);
+    wasmBufferFree(result);
+    return encoded;
+}
+
+// Decode svm data provided in encodedData value and returns a json object of the data
+export function decodeInput(encodedData: any) : JSON {
+    const buf = newWasmBuffer(encodedData);
+    const result = call("wasm_decode_inputdata", buf);
+    const json = loadWasmBufferDataAsJson(result);
+    wasmBufferFree(buf);
+    wasmBufferFree(result);
+    return json;
+}
+
+
+// Encodes binary provided binary data as a hex binary string (without an 0x prefix)
+function binToString(array: Uint8Array) : string {
+    let result = "";
+    for (const b of array) {
+        let s = b.toString(16);
+
+        // padding
+        if (s.length < 2) {
+            s = "0" + s;
+        }
+        result += s;
+    }
+    return result;
+}
+
+
+///// Internal help functions below
+
 // Call an svm_codec function with the provided buffer. Returns result buffer.
-export function call(funcName: string, buf) {
+function call(funcName: string, buf) : Uint8Array {
+    console.log("caaling svm function: " + funcName);
```

Comment:
typo

Reply:
please change to SVM (capital letters)

Reply:
@avive

Reply:
I removed this debug line in the latest commit.